### ISSUE TITLE
version 1.7 does not serve configured certificate

### ISSUE THREAD

**Issue by johnypony3** [containous/traefik#3935]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

Configured traefik with:

* http, https endpoints, with http forwarding to https.
* ssl certificate

### What did you expect to see?

traefik serving the configured certificate

### What did you see instead?

traefic serves the default certificate

### What was your workaroud?

Downgraded to 1.6.6 without any configuration change. traefik acts as expected

### Output of `traefik version`: (_What version of Traefik are you using?_)

1.7

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
...
defaultEntryPoints = ["https", "http"]
rootCAs = [ "/etc/traefik/digicert-ca.crt" ]

[entryPoints]
  [entryPoints.http]
  address = ":80"
    [entryPoints.http.redirect]
    entryPoint = "https"
  [entryPoints.https]
  address = ":443"
    [entryPoints.https.tls]
      [[entryPoints.https.tls.certificates]]
      certFile = "cert/tls.crt"
      keyFile = "cert/tls.key"
...
```

### If applicable, please paste the log output in DEBUG level (`--logLevel=DEBUG` switch)

```
...
time="2018-09-26T18:34:16Z" level=debug msg="Add certificate for domains *.redacted.com,*.redacted.com,*.redacted.com"
...
time="2018-09-26T18:34:43Z" level=debug msg="Serving default cert for request: \"redacted.us-west-2.elb.amazonaws.com\""
...
```

**Comment by Juliens** [comment]
Thanks for reporting this issue.

Could you provide us your logs in debug mode, and your entire configuration? (do you use cluster mode?)

**Comment by johnypony3** [comment]
i am unable to provide logs, we do not use cluster mode, my configuration is:

```toml
logLevel = "INFO"

defaultEntryPoints = ["https", "http"]
rootCAs = [ "/etc/traefik/digicert-ca.crt" ]

[entryPoints]
  [entryPoints.http]
  address = ":80"
    [entryPoints.http.redirect]
    entryPoint = "https"
  [entryPoints.https]
  address = ":443"
    [entryPoints.https.tls]
      [[entryPoints.https.tls.certificates]]
      certFile = "cert/tls.crt"
      keyFile = "cert/tls.key"

[traefikLog]
  format = "common"

[accessLog]
  format = "common"

[api]
  entryPoint = "traefik"
  dashboard = true

[consulCatalog]
  endpoint = "registry:8500"
  watch = true
  prefix = "traefik"
  trace = true
  exposedByDefault = true
```

**Comment by johnypony3** [comment]
after discussing with @Juliens this is expected behavior, although I believe 
https://github.com/containous/traefik/pull/3948 is a great idea.

to override the default certificate, i modified my configuration from:
```yaml
  [entryPoints.https]
  address = ":443"
    [entryPoints.https.tls]
      [[entryPoints.https.tls.certificates]]
      certFile = "cert/tls.crt"
      keyFile = "cert/tls.key"
```
to
```
  [entryPoints.https]
  address = ":443"
    [entryPoints.https.tls]
      [entryPoints.https.tls.defaultCertificate]
      certFile = "cert/tls.crt"
      keyFile = "cert/tls.key"
```

**Comment by traefiker** [comment]
Closed by #3948.
